common/flatpak-dir.c

```diff
@@ -8079,6 +8274,168 @@ flatpak_dir_remote_fetch_summary (FlatpakDir   *self,
   return TRUE;
 }
 
+static FlatpakRemoteState *
+_flatpak_dir_get_remote_state (FlatpakDir   *self,
+                               const char   *remote,
+                               gboolean      optional,
+                               GBytes       *opt_summary,
+                               GBytes       *opt_summary_sig,
+                               GCancellable *cancellable,
+                               GError      **error)
+{
+  g_autoptr(FlatpakRemoteState) state = flatpak_remote_state_new ();
+  g_autoptr(GError) my_error = NULL;
+
+  if (error == NULL)
+    error = &my_error;
```

Comment:
The issue is that some lower level ostree code crashes if error is NULL.

Reply:
It might even be fixed, but not in older versions. I forgot exactly what it was :/

Reply:
How about
```
#if OSTREE_CHECK_VERSION (mumble, mumble)
if (error == NULL)
  error = &my_error;
#endif
```
otherwise this is going to turn into an anti-pattern forever.